providers/implementations/keymgmt/ml_dsa_kmgmt.c.in

```diff
@@ -465,7 +465,11 @@ static void *ml_dsa_gen_init(void *provctx, int selection,
     if ((gctx = OPENSSL_zalloc(sizeof(*gctx))) != NULL) {
         gctx->provctx = provctx;
         if (!ml_dsa_gen_set_params(gctx, params)) {
-            OPENSSL_secure_clear_free(gctx, sizeof(*gctx));
+#ifdef OPENSSL_PEDANTIC_ZEROIZATION
```

Comment:
This change also does not look like it is needed. If you look at the internals of ml_dsa_gen_set_params() it only handles 2 parameters - the seed and a prop query. If the seed part fails it would fail inside the get_string_internal() and this only copies the data as the final step before returning 1.